parquet/src/main/java/org/apache/iceberg/parquet/TypeWithSchemaVisitor.java

```diff
@@ -149,6 +142,29 @@
     }
   }
 
+  private static <T> T visitTwoLevelList(Types.ListType iListType, Types.NestedField iListElement, GroupType pListType,
+      Type pListElement, TypeWithSchemaVisitor<T> visitor) {
+    T elementResult = visitField(iListElement, pListElement, visitor);
+
+    return visitor.list(iListType, pListType, elementResult);
+  }
+
+  private static <T> T visitThreeLevelList(Types.ListType iListType, Types.NestedField iListElement,
+      GroupType pListType, Type pListElement, TypeWithSchemaVisitor<T> visitor) {
+    visitor.fieldNames.push(pListType.getFieldName(0));
+
+    try {
+      T elementResult = null;
+      if (pListElement != null) {
```

Comment:
The `determineListElementType` method doesn't perform the field count check and return null, so this is always true. I think you may have thought that `List.get` would return null?

Reply:
Sg, thanks